src/error.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+/**
+ * @file error.h
+ * @author krithigasr
+ * @brief Header file for error types
+ * Any non zero value should be considered as error
+ * @version 0.1
+ * @date 2020-11-29
+ * 
+ * @copyright Copyright (c) 2020
+ * 
+ */
+
+
+typedef enum {
+    ERR_SUCCESS,                //0
```

Comment:
Maybe you can remove ERR_SUCCESS form the enum, or rename it. Since SUCCESS is not err.